### DISCUSSION TITLE
deletePage don't work for build

### DISCUSSION THREAD

**Discussion by serejajar** [gatsbyjs/gatsby#29956] [Help]
## Description

I'm trying to exclude all utils/components files in /pages directory
e.g.: 
src/pages/page-a/components/component.js
src/pages/theme/page-b/components/component.js
src/pages/theme/page-b/utils/index.js
src/pages/page-a/utils/index.js
/pages/utils.js
and etc.

```
exports.onCreatePage = ({ page, actions }) => {
  const { deletePage } = actions;
  const exclude = ['utils', 'components'];
  const isntPage = exclude.findIndex(str => page.path.includes(str)) !== -1;

  if (isntPage) {
    deletePage(page);
  }
}

```
I use the onCreatePage action for it and it works as expected only in develop mode. Build will show the error:

> A page component must export a React component for it to be valid. Please make sure this file exports a React component:
> ../src/pages/page-a/utils/index.js

### Steps to reproduce

I've create a sample repo where you can easily reproduce the bug. See https://github.com/serejajar/gatsbyjs-bug-with-deletePage

### Expected result

I had expected that utils files will be excluded using the deletePage for build too.

### Actual result
As outlined in the description, the GatsbyJS trying to create pages from the utils files.

### Environment
```

  System:
    OS: Linux 4.15 Ubuntu 16.04.7 LTS (Xenial Xerus)
    CPU: (2) x64 AMD Athlon(tm) II X2 245 Processor
    Shell: 4.3.48 - /bin/bash
  Binaries:
    Node: 12.14.0 - ~/.nvm/versions/node/v12.14.0/bin/node
    Yarn: 1.22.5 - ~/.nvm/versions/node/v12.14.0/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.14.0/bin/npm
  Languages:
    Python: 2.7.12 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.182
    Firefox: 85.0.1
  npmPackages:
    gatsby: ^2.32.8 => 2.32.8 
    gatsby-image: ^2.11.0 => 2.11.0 
    gatsby-plugin-gatsby-cloud: ^1.0.2 => 1.0.2 
    gatsby-plugin-manifest: ^2.12.1 => 2.12.1 
    gatsby-plugin-offline: ^3.10.2 => 3.10.2 
    gatsby-plugin-react-helmet: ^3.10.0 => 3.10.0 
    gatsby-plugin-sharp: ^2.14.3 => 2.14.3 
    gatsby-source-filesystem: ^2.11.1 => 2.11.1 
    gatsby-transformer-sharp: ^2.12.0 => 2.12.0 
  npmGlobalPackages:
    gatsby-cli: 2.19.2

```

**Comment by LekoArts** [comment]
Hi! Thanks for the issue.

Are you sure that the pages are actually created? Gatsby will print the warning even when the pages are not created in the end. You can check this by looking into your `public` folder after building.

**Comment by serejajar** [comment]
@LekoArts you right, the `public` doesn't contain excluded pages. How can I fix the problem? Except adding a component to all utils files.

**Comment by LekoArts** [comment]
Ah, now I understand the issue. You have two options then:

1) Move everything that is not a page outside of `src/pages`
2) Overwrite the internal `gatsby-plugin-page-creator` by adding it to your gatsby-config.js and exclude the files/folders: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-page-creator
